lib/SDCard/SDCard.h

```diff
@@ -7,15 +7,29 @@
 
 #include <SD.h>
 
+#define FILENAME "/ser4data.csv"
+
 class SDCard {
 private:
-  bool inited;
+  bool mounted;
+  void printDirectory(File dir, int numTabs);
+  File dataFile;
 
 public:
-  File dataFile;
+  bool logEnabled;
   void init();
   void re_init();
-  bool isInited() { return inited; }
+  bool isMounted() { return mounted; }
+  bool isReadyForLog() { return mounted && dataFile != 0; }
+  // write a string into the dataFile
   void write(string msg);
+  // prints the directory tree of the card
```

Comment:
I still do not get how the current implementation prevents (or properly detects the removal&re-insertion of the sd card) or more generally checks that is is able to properly write (for the display indication)?